pkg/command/bundle.go

```diff
@@ -47,14 +51,39 @@ var _ Interface = (*BundleOptions)(nil)
 func (opts *BundleOptions) AddFlags(cmd *cobra.Command) {
 	cmd.Flags().String("bundle", "", "Where to publish the bundle.")
 	cmd.Flags().String("directory", ".", "The directory to bundle up.")
+
+	cmd.Flags().String("git-url", "", "The git URL to clone the source from if using git clone rather than a bundle image (e.g. if using mink inside a CI/CD pipeline).")
+	cmd.Flags().String("git-rev", "", "The git revision (branch, tag, SHA) to clone the source from if using git clone rather than a bundle image (e.g. if using mink inside a CI/CD pipeline).")
+	cmd.Flags().Bool("git-verbose", false, "If using git to clone the source enable verbose logging")
 }
 
 // Validate implements Interface
 func (opts *BundleOptions) Validate(cmd *cobra.Command, args []string) error {
 	viper.BindPFlags(cmd.Flags())
+
 	opts.ImageName = viper.GetString("bundle")
 	opts.Directory = viper.GetString("directory")
 
+	gitURL := viper.GetString("git-url")
+	if gitURL != "" {
+		if opts.GitLocation == nil {
+			opts.GitLocation = &source.GitLocation{}
+		}
+		opts.GitLocation.URL = gitURL
+		opts.GitLocation.Revision = viper.GetString("git-rev")
+		opts.GitLocation.Verbose = viper.GetBool("git-verbose")
+
+		// lets create a sample source bundle image tag...
+		var err error
+		opts.tag, err = name.NewTag("gcr.io/sample/source-bundle:latest", name.WeakValidation)
+		if err != nil {
+			return err
+		}
+	}
+
+	if opts.GitLocation != nil {
+		return nil
+	}
```

Comment:
I think that generally what this change is introducing is the idea that there are multiple modes of bundles.  What we have today is the kontext-based upload of local sources, and this adds Git.

I think that it would be useful to add a stronger sense of "mode" to BundleOptions as an explicit thing.  I feel like with this strong sense of "mode" in BundleOptions, we could replace `source.CreateSourceSteps` with a method on BundleOptions that keyed off of the mode, instead of passing everything as a growing argument list to the source package 🤔